src/rigid_pendulum_poc/generalized_alpha_time_integrator.cpp

```diff
@@ -106,111 +109,125 @@ std::tuple<State, HostView1D> GeneralizedAlphaTimeIntegrator::AlphaStep(
             "Iteration number: " + std::to_string(time_stepper_.GetNumberOfIterations() + 1) + "\n"
         );
 
-        auto residuals = ComputeResiduals(gen_coords, vector);
-        // TODO: Provide actual force increments
-        auto increments = residuals;
+        auto gen_coords_next = ComputeUpdatedGeneralizedCoordinates(gen_coords, x);
 
-        if (this->CheckConvergence(residuals, increments)) {
+        // Compute the residuals and check for convergence
+        auto residuals = ComputeResiduals(acceleration, mass_matrix, gen_forces);
+        if (this->CheckConvergence(residuals)) {
             this->is_converged_ = true;
             break;
         }
 
+        // Compute the iteration matrix and solve the linear system to get the increments
         auto iteration_matrix = ComputeIterationMatrix(gen_coords, matrix);
-        auto gen_coords_delta = residuals;
-        auto constraints_delta = constraints;
-        solve_linear_system(iteration_matrix, gen_coords_delta);
-        solve_linear_system(iteration_matrix, constraints_delta);
+        auto delta_x = residuals;
+        auto delta_constraints = constraints;
 
+        solve_linear_system(iteration_matrix, delta_x);
+        solve_linear_system(iteration_matrix, delta_constraints);
+
+        // Update the velocity, acceleration, and constraints based on the increments
         Kokkos::parallel_for(
             size,
             KOKKOS_LAMBDA(const int i) {
-                gen_coords(i) += gen_coords_delta(i);
-                gen_velocity(i) += gamma_prime * gen_coords_delta(i);
-                gen_accln(i) += beta_prime * gen_coords_delta(i);
-                constraints(i) += constraints_delta(i);
+                x(i) += delta_x(i);
+                velocity(i) += GAMMA_PRIME * delta_x(i);
+                acceleration(i) += BETA_PRIME * delta_x(i);
+                constraints(i) += delta_constraints(i);
```

Comment:
I don't think we want to call this variable constraints; maybe lagmult or something short for LagrangeMultiplier.